docs/assets/diagrams/linking/2-7-request-challenge.puml

```diff
@@ -0,0 +1,102 @@
+@startuml
+
+title: Request Challenge
+
+hide footbox
+
+box "Mobile Device" #Light
+  actor Alice
+  participant "PISP App" as App
+end box
+box "PISP" #Light
+  participant "Firebase" as Firebase
+  participant "PISP Server" as Server
+end box
+box "Mojaloop" #Light
+  participant "Switch" as Mojaloop
+end box
+
+'start flow
+activate Server
+Server -> Mojaloop ++: ""**POST /consents/123/generateChallenge**""
+Server <-- Mojaloop: ""202 Accepted""
+deactivate Server
+
+Mojaloop -> Mojaloop: Generate a\nnew challenge
+
+rnote left of Mojaloop #Light
+  ""{""
```

Comment:
Something to potentially take into consideration. Aren't there technically 2 separate steps: 
1. requesting and signing the challenge for registration
2. requesting and signing the challenge for authentication

Reply:
sorry can you help to give more context here? in linking we only sign the challenge for consent registration right?
the other signing process is in the transfer flow where we sign the transaction quote to give authorization

Reply:
yeah you're right, this is just for consent registration so this is fine. Ignore what I mentioned above.